rpc/core/net.go

```diff
@@ -172,9 +172,12 @@ func NetInfo(ctx *rpctypes.Context) (*ctypes.ResultNetInfo, error) {
 	// TODO: Should we include PersistentPeers and Seeds in here?
 	// PRO: useful info
 	// CON: privacy
+
+	nodeInfo := p2pTransport.NodeInfo().(p2p.DefaultNodeInfo)
+
 	return &ctypes.ResultNetInfo{
 		Listening: p2pTransport.IsListening(),
-		Listeners: p2pTransport.Listeners(),
+		Listeners: []string{fmt.Sprintf("Listener(%v@%v)", nodeInfo.ListenAddress, nodeInfo.ExternalAddress)},
```

Comment:
I thought I'd add the `ListenAddress` here, but if you would prefer to not make a technically breaking change to the RPC we could keep exact compatibility here and just return external address or listen address when external not set.